File: app/src/main/java/com/udacity/sandwichclub/utils/JsonUtils.java
package com.udacity.sandwichclub.utils;

import com.udacity.sandwichclub.model.Sandwich;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class JsonUtils {

    public static Sandwich parseSandwichJson(String json) throws JSONException {
        if (json == null || json.isEmpty()) {
            return null;
        }

        JSONObject sandwichJsonObject = new JSONObject(json);
        JSONObject nameJsonObject = sandwichJsonObject.getJSONObject("name");

        String mainName = nameJsonObject.getString("mainName");
        List<String> alsoKnownAs = getArrayValues(nameJsonObject.getJSONArray("alsoKnownAs"));
        String placeOfOrigin = sandwichJsonObject.getString("placeOfOrigin");
        String description =  sandwichJsonObject.getString("description");
        String image = sandwichJsonObject.getString("image");
        List<String> ingredients = getArrayValues(sandwichJsonObject.getJSONArray("ingredients"));

        return new Sandwich(mainName, alsoKnownAs, placeOfOrigin, description, image, ingredients);
    }

    private static List<String> getArrayValues(JSONArray jsonArray) throws JSONException {
        List<String> values = new ArrayList<>();

        if (jsonArray != null && jsonArray.length() > 0) {
            for (int i = 0; i < jsonArray.length(); i++) {
                values.add(jsonArray.getString(i));
            }
        }
        return values;
    }
}

File: app/src/main/java/com/udacity/sandwichclub/utils/JsonUtils.kt
package com.udacity.sandwichclub.utils

import com.udacity.sandwichclub.model.Sandwich
import org.json.JSONArray
import org.json.JSONObject

object JsonUtils {

    fun parseSandwichJson(json: String): Sandwich? {
        if (json.isBlank()) {
            return null
        }

        val sandwichJsonObject = JSONObject(json)
        val nameJsonObject = sandwichJsonObject.getJSONObject("name")

        val mainName = nameJsonObject.getString("mainName")
        val alsoKnownAs = getArrayValues(nameJsonObject.getJSONArray("alsoKnownAs"))
        val placeOfOrigin = sandwichJsonObject.getString("placeOfOrigin")
        val description =  sandwichJsonObject.getString("description")
        val image = sandwichJsonObject.getString("image")
        val ingredients = getArrayValues(sandwichJsonObject.getJSONArray("ingredients"))

        return Sandwich(mainName, alsoKnownAs, placeOfOrigin, description, image, ingredients)
    }

    private fun getArrayValues(jsonArray: JSONArray?): MutableList<String>? {
        val values = mutableListOf<String>()

        if (jsonArray != null && jsonArray.length() > 0) {
            for (it in 0..(jsonArray.length() - 1)) {
                values.add(jsonArray[it].toString())
            }
        }
        return values
    }
}
